### ISSUE TITLE
The use of ArrayList in the Journey class is prone to thread concurrency issues

### ISSUE THREAD

**Issue by nickp10** [anotheria/moskito#263]
The ArrayList in the Journey class is causing thread concurrency issues. If you note the basic example from this Stack Overflow thread:

https://stackoverflow.com/questions/38531332/why-am-getting-null-value-from-arraylist-after-adding-value-using-thread

You'll note that `null` entries can mistakenly get added to the ArrayList. When this happens, the Journey's removeStepByName will throw a NullPointerException when it goes to iterate the list and comes across this `null` entry.

**Comment by dvayanu** [comment]
Hello, thank you for the issue and the pull request, we will review it shortly. Is it something you encountered in production?

**Comment by nickp10** [comment]
Yes this is a problem I'm currently experiencing. I have a simple GET request that has a tracer enabled on it. I'm using JMeter to send about 100 concurrent requests to this GET request and I'm getting a NullPointerException on average once each time I run my JMeter test.

![image](https://user-images.githubusercontent.com/13040092/221627737-fd59fea6-c2ca-44f5-a0a3-79af155178c8.png)

**Comment by dvayanu** [comment]
is any action required with this issue?

**Comment by nickp10** [comment]
This can be closed.
